src/back-end/lib/routers/auth.ts

```diff
@@ -382,14 +382,47 @@ async function establishSessionWithClaims(
   request: Request<any, Session>,
   tokenSet: TokenSet
 ) {
+  /**
+   * With current OIDC instance, returns an object similar to the following:
+   * @example (githubpublic)
+   * ```json
+   * {
+   *   exp: ,
+   *   iat: ,
+   *   auth_time: ,
+   *   jti: '',
+   *   iss: '',
+   *   aud: '',
+   *   sub: '',
+   *   typ: '',
+   *   azp: '',
+   *   nonce: '',
+   *   session_state: '',
+   *   at_hash: '',
+   *   sid: '',
+   *   identity_provider: '',
+   *   org_verified: '',
+   *   email_verified: ,
+   *   name: '',
+   *   github_username: '',
+   *   github_id: '',
+   *   preferred_username: '',
+   *   orgs: '',
+   *   given_name: '',
+   *   display_name: '',
+   *   family_name: '',
+   *   email: ''
+   *  }
+   * ```
+   */
   const claims = tokenSet.claims();
   let userType: UserType;
-  const identityProvider = getString(claims, "loginSource");
+  const identityProvider = getString(claims, "identity_provider");
   switch (identityProvider) {
-    case GOV_IDP_SUFFIX.toUpperCase():
+    case GOV_IDP_SUFFIX:
       userType = UserType.Government;
       break;
-    case VENDOR_IDP_SUFFIX.toUpperCase():
+    case VENDOR_IDP_SUFFIX:
```

Comment:
Are the `*_IDP_SUFFIX`es guaranteed to be lowercase?

Reply:
it's guaranteed on the application side via https://github.com/bcgov/digital_marketplace/blob/development/src/shared/config.ts#L17 and on the SSO side, it's changed from coming in as: 

```
{
...
  loginSource: 'IDIR'
}
``` 

to: 

```
{
...
identity_provider: 'idir'
}